### PR TITLE
[JIT] dont optimize device dtype on inline

### PR DESCRIPTION
**Pull request by eellison**
Follow up to https://github.com/pytorch/pytorch/pull/36404

Adding prim::device and prim::dtype to list of skipped peepholes when we run inlining. In the long term another fix may not be to encode shape / dtype info on the traced graph, because it is not guaranteed to be correct. This is blocked by ONNX currently.

Partial fix for https://github.com/pytorch/pytorch/issues/43134

### CODE REVIEW

File: test/jit/test_tracer.py [hunk 1]
```diff
@@ -1188,6 +1188,17 @@ def tensor_size(x: torch.Tensor) -> torch.Tensor:
             torch.tensor([15])
         )
 
+        @torch.jit.script
+        def use_device(x):
+            return torch.zeros_like(x, device=x.device)
```
Comment (wanchaol) [inline test/jit/test_tracer.py:1193]: zeros_like will default the device to the input, maybe use `zero` to test instead?

File: test/jit/test_tracer.py [hunk 2]
```diff
@@ -1188,6 +1188,17 @@ def tensor_size(x: torch.Tensor) -> torch.Tensor:
...
+
+        def foo(x):
+            return use_device(x)
+
+        traced_tensor_size = torch.jit.trace(foo, torch.rand(7,))
+        self.run_pass('inline', traced_tensor_size.graph)
+        FileCheck().check("aten::zeros_like").run(traced_tensor_size.graph)
```
Comment (wanchaol) [inline test/jit/test_tracer.py:1200]: shouldn't we also check if there's a `prim::device` node?

**Review by wanchaol — APPROVED** [review]
Looks good, thanks for the fix!

**Comment by neginraoof** [comment]
Hey @eellison 
I'm also unable to repro the segfault, but I'm seeing some failures with tests like faster_rcnn, etc. which are failing due to the change in Device node.

To fix those failures, we need to updated the logic for exporting prim ops are parsed. As the device is updated from:
prim::Constant('device') to prim::device

This code is in torch/onnx/utils.py in _run_symbolic_function.
The line below should be added as a new else if check for prim ops:

```python
        elif ns == "prim":
            if op_name == "Constant" and not n.mustBeNone():
               ...
            elif n.mustBeNone() or op_name == "ListConstruct" or op_name == "ListUnpack":
               ...
            ....
            elif op_name == "device" and n.output().type().kind() == "DeviceObjType":
                return None
```

**Pull request closed without merging**